tests/Behat/FeatureContext.php

```diff
@@ -91,4 +92,73 @@ protected function assertElementType(NodeElement $element, string $type): void {
     }
   }
 
+  /**
+   * Checks the given node with title has the number of revisions and states.
+   *
+   * // phpcs:disable
+   * @Then the node :title should have :number number of revisions with the following states:
+   * | state 1 |
+   * | state 2 |
+   * |   ...   |
+   * // phpcs:enable
+   */
+  public function theNodeShouldHaveNumberForRevisionsWithTheFollowingStates(string $title, int $number, TableNode $options) {
+    $node = $this->getNodeByTitle($title);
+    /** @var \Drupal\Core\Entity\ContentEntityStorageInterface $storage */
+    $storage = \Drupal::entityTypeManager()->getStorage($node->getEntityTypeId());
+    $rids = $storage->revisionIds($node);
+
+    if (count($rids) !== $number) {
+      throw new ExpectationException('The number of revision doesn\'t match. Expected: ' . count($rids), $this->getSession());
+    }
+
+    // Retrieve the options table from the test scenario and flatten it.
+    $expected_states = $options->getRows();
+    array_walk($expected_states, function (&$value) {
+      $value = reset($value);
+    });
+
+    /** @var \Drupal\content_moderation\ModerationInformationInterface $moderation_info */
+    $moderation_info = \Drupal::service('content_moderation.moderation_information');
+    /** @var \Drupal\workflows\WorkflowInterface $workflow */
+    $workflow = $moderation_info->getWorkflowForEntity($node);
+    /** @var \Drupal\workflows\WorkflowTypeInterface $workflow_plugin */
+    $workflow_plugin = $workflow->getTypePlugin();
+
+    // Retrieve the states from all the revisions and flatten it.
+    $revision_states = [];
+    foreach ($rids as $rid) {
+      $revision = $storage->loadRevision($rid);
+      $revision_states[] = $workflow_plugin->getState($revision->moderation_state->value)->label();
+    }
+
+    Assert::assertEquals($expected_states, $revision_states);
+  }
+
+  /**
+   * Retrieves a node by its title.
+   *
+   * @param string $title
+   *   The node title.
+   *
+   * @return \Drupal\node\NodeInterface
+   *   The node entity.
+   */
+  protected function getNodeByTitle(string $title): NodeInterface {
```

Comment:
Does it really make sense to have this short function? $storage->loadByProperties does pretty much everything we need here.

Reply:
We still would need to add the content of the function to the step, that would make it bigger. I like to keep this in a separate function to keep things clean.